pkg/services/correlations/api.go

```diff
@@ -56,8 +56,8 @@ func (s *CorrelationsService) createHandler(c *contextmodel.ReqContext) response
 			return response.Error(http.StatusNotFound, "Data source not found", err)
 		}
 
-		if errors.Is(err, ErrSourceDataSourceReadOnly) {
-			return response.Error(http.StatusForbidden, "Data source is read only", err)
+		if errors.Is(err, ErrCorrelationsReadOnly) {
+			return response.Error(http.StatusForbidden, "Correlations is read only", err)
```

Comment:
how can this happen? i'm creating a new one, how can i get a read-only error?

Reply:
Oh, true that! Removed the check in 1806646de5837faebac97526eff1cf82dac90daa. Is there a neat way to lint or annotate such cases? Something that would work like good old `try { } catch(ErrorA a) {} catch(ErrorB b) {}`